src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesImplementation.OpenSsl.cs

```diff
@@ -35,6 +39,56 @@ private static OpenSslCipherLite CreateLiteCipher(
             return new OpenSslCipherLite(algorithm, blockSize, paddingSize, key, iv, encrypting);
         }
 
+        protected override void EncryptKeyWrapPaddedCore(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            int written = KeyWrap(source, destination, enc: 1);
+            Debug.Assert(written == destination.Length);
+        }
+
+        protected override int DecryptKeyWrapPaddedCore(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            return KeyWrap(source, destination, enc: 0);
+        }
+
+        private int KeyWrap(ReadOnlySpan<byte> source, Span<byte> destination, int enc)
+        {
+            Debug.Assert(enc is 0 or 1);
+
+            SafeEvpCipherCtxHandle ctx = GetKey().UseKey(
+                state: enc,
+                static (enc, key) =>
+                {
+                    int keySizeInBits = key.Length * 8;
+
+                    IntPtr algorithm = GetKeyWrapAlgorithm(keySizeInBits);
+
+                    return Interop.Crypto.EvpCipherCreate(
+                        algorithm,
+                        ref MemoryMarshal.GetReference(key),
+                        key.Length * 8,
+                        ref MemoryMarshal.GetReference(ReadOnlySpan<byte>.Empty),
+                        enc);
+                });
+
+            int written;
+
+            using (ctx)
+            {
+                bool ret = Interop.Crypto.EvpCipherUpdate(
```

Comment:
We don't need to call `EVP_CipherFinal_ex`? If not it would be good to leave a comment explaining why, or at least that it is intentionally omitted.